### PR TITLE
Update almalinux-deploy.sh for md raid support for ESP partitions

### PR DESCRIPTION
**Pull request by jonasf21**
Add software md raid support for ESP partitions, script will break when used on software raid (1) ESP partitions, fixing issue #97

### CODE REVIEW

**Comment by matthewcky2k** [comment]
Any updates on the PR?

**Comment by pizzi80** [comment]
any update?

**Comment by soksanichenko** [comment]
Hi. I'll review today or on Monday. Thank you for reminder.

**Comment by soksanichenko** [comment]
@jonasf21 Hi. Please check the review. Thank you!

**Comment by jonasf21** [comment]
@soksanichenko Ok, aarch64 support is now included, please have a look.

File: almalinux-deploy.sh [hunk 1]
```diff
@@ -754,12 +754,46 @@ add_efi_boot_record() {
...
+        disk_name="$(echo "${device}" | sed -re 's/(p|)[0-9]$//g')"
+        disk_num="$(echo "${device}" | tail -c 2|sed 's|[^0-9]||g')"
+    fi
+        
+    if [[ ${arch} == "x86_64" && $device == *"/dev/md"* ]]; then
+        efibootmgr -c -L "AlmaLinux" -l "\EFI\almalinux\shimx64.efi" -d "${disk_name1}" -p "${disk_num1}"
+        efibootmgr -c -L "AlmaLinux" -l "\EFI\almalinux\shimx64.efi" -d "${disk_name2}" -p "${disk_num2}"
+        
+    elif [[ ${arch} == "aarch64" && $device == *"/dev/md"* ]]; then
+    	  efibootmgr -c -L "AlmaLinux" -l "\EFI\almalinux\grubaa64.efi" -d "${disk_name1}" -p "${disk_num1}"
```
Comment (andrewlukoshko) [inline almalinux-deploy.sh:785]: why \EFI\almalinux\grubaa64.efi is used?
maybe \EFI\almalinux\shimaa64.efi ?

File: almalinux-deploy.sh [hunk 2]
```diff
@@ -754,12 +754,46 @@ add_efi_boot_record() {
...
+        disk_name1="$(echo "${soft_dev1}" | sed -re 's/(p|)[0-9]$//g')"
+        disk_name2="$(echo "${soft_dev2}" | sed -re 's/(p|)[0-9]$//g')"
+        disk_num1="$(echo "${soft_dev1}" | tail -c 2|sed 's|[^0-9]||g')"
+        disk_num2="$(echo "${soft_dev2}" | tail -c 2|sed 's|[^0-9]||g')"
+    else
+        disk_name="$(echo "${device}" | sed -re 's/(p|)[0-9]$//g')"
+        disk_num="$(echo "${device}" | tail -c 2|sed 's|[^0-9]||g')"
+    fi
+        
+    if [[ ${arch} == "x86_64" && $device == *"/dev/md"* ]]; then
```
Comment (andrewlukoshko) [inline almalinux-deploy.sh:780]: Where ${arch} comes from?

**Review by andrewlukoshko — CHANGES_REQUESTED** [review]
Please check comments

File: almalinux-deploy.sh (hunk 1 above)
Comment (jonasf21) [reply almalinux-deploy.sh:785]: I dont have a aarch64 system to test. But it appears you are right, it should be shimaa64.efi now - i'll fix.

File: almalinux-deploy.sh (hunk 2 above)
Comment (jonasf21) [reply almalinux-deploy.sh:782]: fixed

**Comment by matthewcky2k** [comment]
Any update on this PR?

**Comment by andrewlukoshko** [comment]
Sorry for delay, I'm going to merge this soon.
Need some extra testing.

**Pull request merged**